### DISCUSSION TITLE
.NET6 Lambda Test Tool hangs when running Debug with Rider

### DISCUSSION THREAD

**Discussion by nrkirby** [aws/aws-lambda-dotnet#1163] [Q&A] [answered]
### Describe the bug

In a simple Lambda project I have I can run the test tool without a debugger fine:

![image](https://user-images.githubusercontent.com/6445534/165896788-42a6cb40-6a4d-4bcb-8ce4-39d1a94cbc60.png)

When I run it with Debug (in Rider) the test tool hangs and doesn't make any progress:

![image](https://user-images.githubusercontent.com/6445534/165896823-ad7f7588-a36f-4493-95b8-7acc73ec9309.png)

### Expected Behavior

The test tool runs in Debug.

### Current Behavior

The test tool hangs in Debug.

### Reproduction Steps

I've created an example project [here](https://github.com/NRKirby/LambdaTestToolExample) which demonstrates the issue I am having. 

Note that I've created a [launchSettings.json](https://github.com/NRKirby/LambdaTestToolExample/blob/main/src/Example.Api/Properties/launchSettings.json) file for starting the debug tool.

### AWS .NET SDK and/or Package version used

Amazon.Lambda.Core 2.1.0
Amazon.Lambda.APIGatewayEvents 2.4.0

### Targeted .NET Platform

.NET6

### Operating System and version

Windows 10

**Comment by ashishdhingra** [comment, marked as answer]
Hi @NRKirby,

Good morning.

Unfortunately, I'm unable to reproduce the issue, Lambda Test Tool with JetBrains Rider works fine on my system (refer the below screenshot).
<img width="2221" alt="Screen Shot 2022-04-29 at 10 12 55 AM" src="https://user-images.githubusercontent.com/67916761/165991970-b14d7aab-f395-4056-9058-851f01d105ea.png">

Have you followed the steps at https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#configure-for-jetbrains-rider properly for JetBrains Rider configuration. Kindly note that the Exe path for .NET 6.0 test tool is `/Users/<<username>>/.dotnet/tools/.store/amazon.lambda.testtool-6.0/0.12.1/amazon.lambda.testtool-6.0/0.12.1/tools/net6.0/any/Amazon.Lambda.TestTool.BlazorTester.dll` assuming version `0.12.1` of the test tool is installed.

This appears to be your environmental issue. You may try uninstalling the Lambda Test Tool and reinstalling the latest version (more instructions at https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#installing-and-running).

**NOTE:** I tested this on Mac, not on Windows, bur steps are almost similar.

Thanks,
Ashish

**Reply by ashishdhingra** [reply]
This works fine on Windows as well:
<img width="1290" alt="Screen Shot 2022-04-29 at 11 08 36 AM" src="https://user-images.githubusercontent.com/67916761/165999579-7968c65f-6081-4cbb-8011-ddb4ccac554c.png">

**Comment by olegpuz** [comment]
Always Reproduces for my dev team (10ppl).
What we found working is Run, then Attach Debugger->pick mockTambdaTestTool.exe
Also make sure your profile`s working directory=. (unlike how Visual Studio demands it being a proj dir)
